x/ibc/04-channel/types/msgs.go

```diff
@@ -0,0 +1,379 @@
+func (msg MsgChannelCloseInit) Route() string {
+	return ibctypes.RouterKey
+}
+
+// Type implements sdk.Msg
+func (msg MsgChannelCloseInit) Type() string {
+	return "channel_close_init"
+}
+
+// ValidateBasic implements sdk.Msg
+func (msg MsgChannelCloseInit) ValidateBasic() sdk.Error {
+	if err := host.DefaultPortIdentifierValidator(msg.PortID); err != nil {
+		return sdk.NewError(host.IBCCodeSpace, 1, fmt.Sprintf("invalid port ID: %s", err.Error()))
+	}
+	if err := host.DefaultChannelIdentifierValidator(msg.ChannelID); err != nil {
+		return sdk.NewError(host.IBCCodeSpace, 1, fmt.Sprintf("invalid channel ID: %s", err.Error()))
+	}
+	// Signer can be empty
+	return nil
+}
+
+// GetSignBytes implements sdk.Msg
+func (msg MsgChannelCloseInit) GetSignBytes() []byte {
+	return sdk.MustSortJSON(SubModuleCdc.MustMarshalJSON(msg))
+}
+
+// GetSigners implements sdk.Msg
+func (msg MsgChannelCloseInit) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Signer}
+}
+
+var _ sdk.Msg = MsgChannelCloseConfirm{}
+
+type MsgChannelCloseConfirm struct {
+	PortID      string            `json:"port_id"`
+	ChannelID   string            `json:"channel_id"`
+	ProofInit   commitment.ProofI `json:"proof_init"`
+	ProofHeight uint64            `json:"proof_height"`
+	Signer      sdk.AccAddress    `json:"signer"`
+}
+
+// NewMsgChannelCloseConfirm creates a new MsgChannelCloseConfirm instance
+func NewMsgChannelCloseConfirm(
+	portID, channelID string, proofInit commitment.ProofI, proofHeight uint64,
+	signer sdk.AccAddress,
+) MsgChannelCloseConfirm {
+	return MsgChannelCloseConfirm{
+		PortID:      portID,
+		ChannelID:   channelID,
+		ProofInit:   proofInit,
+		ProofHeight: proofHeight,
+		Signer:      signer,
+	}
+}
+
+// Route implements sdk.Msg
+func (msg MsgChannelCloseConfirm) Route() string {
+	return ibctypes.RouterKey
+}
+
+// Type implements sdk.Msg
+func (msg MsgChannelCloseConfirm) Type() string {
+	return "channel_close_confirm"
+}
+
+// ValidateBasic implements sdk.Msg
+func (msg MsgChannelCloseConfirm) ValidateBasic() sdk.Error {
+	if err := host.DefaultPortIdentifierValidator(msg.PortID); err != nil {
+		return sdk.NewError(host.IBCCodeSpace, 1, fmt.Sprintf("invalid port ID: %s", err.Error()))
+	}
+	if err := host.DefaultChannelIdentifierValidator(msg.ChannelID); err != nil {
+		return sdk.NewError(host.IBCCodeSpace, 1, fmt.Sprintf("invalid channel ID: %s", err.Error()))
+	}
+	if msg.ProofInit == nil {
+		return ErrInvalidChannelProof(DefaultCodespace, "cannot submit an empty proof")
+	}
+	if msg.ProofHeight == 0 {
+		return ErrInvalidChannelProof(DefaultCodespace, "proof height must be > 0")
+	}
+	// Signer can be empty
+	return nil
+}
+
+// GetSignBytes implements sdk.Msg
+func (msg MsgChannelCloseConfirm) GetSignBytes() []byte {
+	return sdk.MustSortJSON(SubModuleCdc.MustMarshalJSON(msg))
+}
+
+// GetSigners implements sdk.Msg
+func (msg MsgChannelCloseConfirm) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Signer}
+}
```

Comment:
`MsgRecvPacket` should be defined here, used for all type of packets

Reply:
(no need to be address in this pr)

Reply:
blocked on #5230